numpy/core/tests/test_datetime.py

```diff
@@ -209,6 +208,26 @@ def test_datetime_scalar_construction(self):
         assert_equal(np.datetime64('1920-03-13T18:33:12.5', 'Y'),
                      np.datetime64('1920'))
 
+        # Construction from numpy types
+        assert_array_equal(np.datetime64(np.int(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.int32(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.int64(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.uint(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.uint32(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.uint64(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.long(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.longlong(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
+        assert_array_equal(np.datetime64(np.ulonglong(1262304000), 's'),
+                     np.datetime64(1262304000, 's'))
```

Comment:
Similarly, you're just repeating yourself here by testing `ulonglong` etc, since `ulonglong is uint64`. Stick to just the fixed-width types.